reference/reference.md

```diff
@@ -0,0 +1,273 @@
+| *extraInitialState?* | *any* | *the (optional) initial state* |
+
+**Returns** [IModelState](#imodelstate)
+
+<br>
+<br>
+
+<!-- buildFeatureState -->
+`buildFeatureState` - Builds the Ngrx state for an entity that is part of a feature module.
+
+```typescript
+buildFeatureState<TState extends IEntityState<TModel>, TParentState, TModel>(
+  type: IModelClass<TModel>,
+  featureStateName: NonNullable<string>,
+  selectParentState: MemoizedSelector<object, TParentState>,
+  extraInitialState?: any
+): IModelState<TParentState, TState, TModel>
+```
+
+| Parameter | Type | Description |
+| --------- | ---- | ----------- |
+| type | IModelClass\<TModel\> | the entity class |
+| featureStateName | NonNullable\<string\> | the name of the feature state |
+| selectParentState | MemoizedSelector\<object, TParentState\> | a selector for the entity's parent state |
+| *extraInitialState?* | *any* | *the (optional) initial feature state* |
+
+**Returns** [IModelState](#imodelstate)
+
+
+
+<!-- -->
+<br>
+<br>
+<!-- -->
+
+<!-- Section: Decorators -->
+## **Decorators**
+
+`@Key` - Designates the key property for the entity.
+
+{% code-tabs %}
+{% code-tabs-item title="src/app/models/account.model.ts" %}
+```typescript
+import { Key } from '@briebug/ngrx-auto-entity';
+
+export class Account {
+  @Key id: number;
+  customer_id: number;
+  accountNumber: number;
+  name: string;
+  amount: number;
+}
+```
+
+<!-- -->
+<br>
+<br>
+<!-- -->
+
+<!-- Section: Facade -->
+## **Facades**
+Prefabricated facades come complete with a core set of properties and methods that may be used "out of the box" without any additional work on your part beyond extending the base facade class.
+
+<br>
+
+**Properties (Selectors)** <br>
+Getters that will return an observable containing data on the **Facade** model's entities and state.
+
+| Property | Return Type | Description |
+| -------- | ----------- | ----------- |
+| all$ | Observable\<TModel[]\> | Get all entities as an array. |
+| entities$ | Observable\<IEntityDictionary\<TModel\>\> | Get all entities as a dictionary. |
+| ids$ | Observable\<EntityIdentity[]\> | Get all entity ids as an array. |
+| total$ | Observable\<number\> | Get the number of entities. |
+| current$ | Observable\<TModel\> | Get the currently selected entity. |
+| currentKey$ | Observable\<EntityIdentity\> | Get the currently selected entity's id. |
+| currentSet$ | Observable\<TModel[]\> | Get the currently selected entities as an array. |
+| currentSetKeys$ | Observable\<EntityIdentity[]\> | Get the currently selected entities' ids as an array. |
+| edited$ | Observable\<Partial\<TModel\>\> | TBD |
+| isDirty$ | Observable\<boolean\> | TBD |
+| currentPage$ | Observable\<Page\> | Get the current **Page** if using pagination. |
+| currentRange$ | Observable\<Range\> | Get the current **Range** if using pagination. |
+| totalPageable$ | Observable\<number\> | TBD |
+| isLoading$ | Observable\<boolean\> | Get whether this **Facade** is loading. |

```

Comment:
Instead of Facade, I would say Entity. Entity would be more accurate for each of the isWhatevers.